projects/plugins/boost/app/modules/cache/Boost_File_Cache.php

```diff
@@ -0,0 +1,113 @@
+			'get'         => $_GET, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		);
+		$args     = array_merge( $defaults, $args );
+
+		return $this->path( $args['request_uri'] ) . $this->cache_key( $args ) . '.html';
+	}
+
+	/*
+	 * Outputs the cached page if it exists for the given request, or current request.
+	 *
+	 * @param array $args - an array containing the request_uri, cookies array, and get array representing the request.
+	 * @return bool - false if page was not cached.
+	 */
+	public function get( $args = array() ) {
+		if ( ! $this->is_cacheable() ) {
+			return false;
+		}
+		$defaults = array(
+			'request_uri' => $this->request_uri,
+			'cookies'     => $_COOKIE, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			'get'         => $_GET, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		);
+		$args     = array_merge( $defaults, $args );
+
+		if ( file_exists( $this->cache_filename( $args ) ) ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents, WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo file_get_contents( $this->cache_filename( $args ) ) . '<!-- cached -->';
+			die();
+		}
+		return false;
+	}
+
+	/*
+	 * Creates the cache directory if it doesn't exist.
+	 *
+	 * @param string $path - The path to the cache directory to create.
+	 */
+	private function create_cache_directory( $path ) {
+		if ( ! is_dir( $path ) ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.dir_mkdir_dirname, WordPress.WP.AlternativeFunctions.file_system_operations_mkdir
+			mkdir( $path, 0755, true );
+		}
+	}
+
+	/*
+	 * Saves the output buffer to the cache file for the given request, or current request.
+	 * Then outputs the buffer to the browser.
+	 *
+	 * @param string $data - The output buffer to save to the cache file.
+	 * @return bool - false if page was not cacheable.
+	 */
+	public function set( $data ) {
+		if ( ! $this->is_cacheable() ) {
+			return false;
+		}
+
+		$args = array(
+			'request_uri' => $this->request_uri,
+			'cookies'     => $_COOKIE, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			'get'         => $_GET, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		);
+
+		$cache_filename = $this->cache_filename( $args );
+
+		$this->create_cache_directory( dirname( $cache_filename ) );
+		$tmp_filename = $cache_filename . uniqid( wp_rand(), true ) . '.tmp';
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_file_put_contents
+		file_put_contents( $tmp_filename, $data );
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.rename_rename
+		rename( $tmp_filename, $cache_filename );
+		$this->get( $args );
```

Comment:
Yes, writing it directly was the way I had it working first, but then I changed to using get(), but then I noticed that wasn't the right way to do it. The code should simply return the output buffer, and it will be sent to the browser. 

Writing to a temp file first allows a busy site to serve the cached file to other visitors while a new file is being created. The rename() is atomic, and overwrites the old cache file. This is in the event that multiple people are creating a cache file for the same page at the exact same time.
WP Super Cache has a "rebuild" feature that serves the old cache page while a new one is being built, a feature we should bring here eventually as it helps a ton.